gateway/handlers/forwarding_proxy.go

```diff
@@ -178,3 +197,59 @@ func (f FunctionAsHostBaseURLResolver) Resolve(r *http.Request) string {
 
 	return fmt.Sprintf("http://%s%s:%d", svcName, suffix, watchdogPort)
 }
+
+// TransparentURLPathTransformer passes the requested URL path through untouched.
+type TransparentURLPathTransformer struct {
+}
+
+// Transform returns the URL path unchanged.
+func (f TransparentURLPathTransformer) Transform(r *http.Request) string {
+	return r.URL.Path
+}
+
+// FunctionPathTruncatingURLPathTransformer always truncated the path to "/".
+type FunctionPathTruncatingURLPathTransformer struct {
+}
+
+// Transform always return a path of "/".
+func (f FunctionPathTruncatingURLPathTransformer) Transform(r *http.Request) string {
+	ret := r.URL.Path
+
+	if ret != "" {
+		matcher := functionMatcher.Copy()
+		parts := matcher.FindStringSubmatch(ret)
+		// In the following regex, in the case of a match the r.URL.Path will be at `0`,
+		// the function name at `1` and the rest of the path (the part we are interested in)
+		// at `2`.  For this transformer, all we need to do is confirm it is a function.
+		if 3 == len(parts) {
```

Comment:
Similar code is repeated on 3 places returning different result. Would be better to go in a single method to make the easier support in future.